### PR TITLE
PS-9789: fixed punchoutCatalogsFormat dynamic sub form saving

### PR DESCRIPTION
**Pull request by asmarovydlo**
Ticket: https://spryker.atlassian.net/browse/PS-9789
Epic: https://spryker.atlassian.net/browse/PS-8578

### CODE REVIEW

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/IndexController.php [hunk 1]
```diff
@@ -65,7 +66,7 @@ public function tableAction(): JsonResponse
     public function createAction(Request $request)
     {
         $punchoutCatalogConnectionsForm = $this->getFactory()
-            ->getPunchoutCatalogConnectionForm();
+            ->getPunchoutCatalogConnectionForm(null, $request->request->has(PunchoutCatalogConnectionForm::FORM_NAME));
```
Comment (dmiseev) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/IndexController.php]: I think it makes sense to create a variable for this form.

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/IndexController.php [hunk 2]
```diff
@@ -111,7 +112,10 @@ public function editAction(Request $request)
         }
 
         $punchoutCatalogConnectionEditForm = $this->getFactory()
-            ->getPunchoutCatalogConnectionForm($punchoutCatalogConnectionTransfer);
+            ->getPunchoutCatalogConnectionForm(
+                $punchoutCatalogConnectionTransfer,
+                $request->request->has(PunchoutCatalogConnectionForm::FORM_NAME)
+            );
```
Comment (dmiseev) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/IndexController.php]: I think it makes sense to create a variable for this form.

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/PunchoutCatalogConnectionForm.php [hunk 3]
```diff
@@ -67,6 +70,7 @@ public function configureOptions(OptionsResolver $resolver): void
             static::OPTION_BUSINESS_UNIT_CHOICES,
             static::OPTION_CONNECTION_FORMAT_SUB_FORM_TYPES,
             static::OPTION_CONNECTION_TYPE_SUB_FORM_TYPES,
+            static::OPTION_IS_SUBMITED
```
Comment (dmiseev) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/PunchoutCatalogConnectionForm.php]:
```suggestion
            static::OPTION_IS_SUBMITED,
```

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/PunchoutCatalogConnectionForm.php [hunk 4]
```diff
@@ -206,7 +210,7 @@ protected function addConnectionTypeSubForms(FormBuilderInterface $builder, arra
      *
      * @return void
      */
-    protected function addConnectionFormatDynamicSubFormListener(FormBuilderInterface $builder): void
+    protected function addConnectionFormatDynamicSubFormListener(FormBuilderInterface $builder, array $options): void
```
Comment (dmiseev) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/PunchoutCatalogConnectionForm.php]: annotations?

**Review by DmitryLymarenko — APPROVED** [review]

**Review by dmiseev — APPROVED** [review]

**Review by demkos — APPROVED** [review]

**Pull request merged**